keras_cv/bounding_box/mask_invalid_detections.py

```diff
@@ -73,14 +74,20 @@ def mask_invalid_detections(bounding_boxes, output_ragged=False):
 
     classes = tf.where(mask, classes, -tf.ones_like(classes))
 
+    if confidence is not None:
```

Comment:
Wait, so confidence filtering was just broken?

Reply:
Not the filtering but converting confidence tensors you ragged.  Basically confidence score still had -1s padded on - even in ragged mode